src/bootstrap/src/utils/tarball.rs

```diff
@@ -31,7 +31,7 @@ pub(crate) enum OverlayKind {
 impl OverlayKind {
     fn legal_and_readme(&self) -> &[&str] {
         match self {
-            OverlayKind::Rust => &["COPYRIGHT", "LICENSE-APACHE", "LICENSE-MIT", "README.md"],
+            OverlayKind::Rust => &["COPYRIGHT", "README.md"],
```

Comment:
I wasn't clear which files the `legal_and_readme` stuff ended up in, vs the logic in `bootstrap/src/core/build_steps/dist.rs`.

What's the 'cheapest' `x` command I can run to generate a set of tarballs, like we might host on rustup?

Reply:
The situation with what gets into rustup is a bit complicated, and I myself am not fully sure how are the archives combined in the end :) You can take a look at `x dist rust-src`, which is quite cheap and doesn't contain any custom `share/xxx` logic. If you want to generate what we do on CI, then `x dist extended` should get you close to it, although I wouldn't exactly call it cheap :laughing:

Reply:
OK, I ran `x dist` and then unpacked every tarball it made. I see:

* `./rustc-nightly-aarch64-apple-darwin/rustc/share/doc/rust/licences/MIT.txt`
* `./rust-nightly-aarch64-apple-darwin/rustc/share/doc/rust/licences/MIT.txt`

So, the license texts folder ended up in `rustc-nightly-<target>` and `rust-nightly-<target>` and nowhere else. But, the COPYRIGHT file ended up in:

* `llvm-bitcode-linker-nightly-aarch64-apple-darwin/COPYRIGHT`
* `rust-std-nightly-aarch64-apple-darwin/COPYRIGHT`
* `rust-analysis-nightly-aarch64-apple-darwin/COPYRIGHT`
* `rustc-nightly-aarch64-apple-darwin/COPYRIGHT`
* `rustc-dev-nightly-aarch64-apple-darwin/COPYRIGHT`
* `rust-src-nightly/COPYRIGHT`
* `rust-nightly-aarch64-apple-darwin/COPYRIGHT`
* `rust-docs-json-nightly-aarch64-apple-darwin/COPYRIGHT`
* `rust-docs-nightly-aarch64-apple-darwin/COPYRIGHT`
* `rustc-nightly-src/COPYRIGHT`

Plus the two generated HTML versions in:

* `rustc-nightly-aarch64-apple-darwin/rustc/share/doc/rust/COPYRIGHT.html`
* `rust-nightly-aarch64-apple-darwin/rustc/share/doc/rust/COPYRIGHT.html`

I will try and change it so that anywhere `COPYRIGHT` goes, the MIT and Apache-2.0 license texts also go.